### PR TITLE
Add inclusive fix types

### PR DESCRIPTION
**Pull request by mark-todd**

### CODE REVIEW

File: endgame_simulations/models.py [hunk 1]
```diff
@@ -83,6 +83,9 @@ class ParameterChange(GenericModel, Generic[UpdateParams]):
     params: UpdateParams
 
 
+UpdateParams = TypeVar("UpdateParams", bound=_BaseUpdateParams)
```
Comment (mkcinu) [inline endgame_simulations/models.py]: That's defined already in line 77.

File: endgame_simulations/endgame_simulation.py [hunk 2]
```diff
@@ -249,14 +255,21 @@ def iter_run(
...
         sampling_interval: float | None = None,
         sampling_years: list[float] | None = None,
+        inclusive: bool = False,
     ) -> Iterator[State]:
         while self.simulation.state.current_time < end_time:
             # Invariant: current params are applied at this point
             if self.next_params_index < len(self._param_set):
                 time, next_params = self._param_set[self.next_params_index]
-                next_stop = min(time, end_time)
+                if inclusive:
```
Comment (mkcinu) [inline endgame_simulations/endgame_simulation.py]: If you defined something like `inclusive_adjustment = self.simulation._delta_time if inclusive else 0.0` then you don't need any extra branching beyond what was already there I think.

File: endgame_simulations/models.py (hunk 1 above)
Comment (mark-todd) [reply endgame_simulations/models.py]: Yeah typing bug without it though - says it's a recursive definition

File: endgame_simulations/endgame_simulation.py (hunk 2 above)
Comment (mark-todd) [reply endgame_simulations/endgame_simulation.py]:
```py
inclusive_adjustment = self.simulation._delta_time if inclusive else 0.0
if self.next_params_index < len(self._param_set):
     time, next_params = self._param_set[self.next_params_index]
     next_stop = min(time, end_time + inclusive_adjustment)
else:
     next_stop = end_time + inclusive_adjustment
     next_params = None

**Pull request merged**